pkg/cache/snapshot.go

```diff
@@ -43,18 +43,57 @@ func (s *Snapshot) RemoveWorkload(wl *workload.Info) {
 	delete(cq.Workloads, workload.Key(wl.Obj))
 	updateUsage(wl, cq.Usage, -1)
 	if cq.Cohort != nil {
-		updateUsage(wl, cq.Cohort.Usage, -1)
+		if features.Enabled(features.LendingLimit) {
```

Comment:
We should have unit tests just for these functions. We should test for the transitions from "usage below guaranteed" and "usage above guaranteed" and the reverse, but also to the same states.
I'm not convinced the current implementation works.

However, maybe a simpler approach is:
1. remove the existing "clusterqueue usage from the cohort"
2. `updateUsage(wl, cq.Usage, +-1)`
3. Add the new "clusterqueue usage from cohort"

Reply:
I have added some tests that can prove the current implementation works.
About `cohort.Usage`, I'm still trying to handle it as the usage of LendingLimit. Because if we count the `cq.Usage` in, we need to overwrite the accumulate function.

Reply:
> However, maybe a simpler approach is:

+1, more readable to me. And I think the code is reuseable for Addworkload.

Reply:
Done.